state-chain/primitives/src/chains/assets.rs

```diff
@@ -389,6 +389,13 @@ macro_rules! assets {
 					})
 				}
 			}
+
+			impl scale_info::prelude::ops::Add<AssetMap<crate::AssetAmount>> for AssetMap<crate::AssetAmount> {
+				type Output = AssetMap<crate::AssetAmount>;
+				fn add(self, other: AssetMap<crate::AssetAmount>) -> AssetMap<crate::AssetAmount>{
+					AssetMap::from_fn(|asset| self.index(asset).saturating_add(*other.index(asset)) )
```

Comment:
Since this is inside the macro definition we can build the AssetMap directly instead of relying on `from_fn`. We can also implement the trait generically for an `N: Saturating`.

Reply:
I'll push this change, I wanted to see if it was possible so have basically already written it.